From c59b84ca3bda8e4244d47901b6966f28dd675434 Mon Sep 17 00:00:00 2001
From: Andrei Dinu <andrei.adrianx.dinu@intel.com>
Date: Thu, 23 May 2013 15:12:23 +0300
Subject: [PATCH] added yocto-standard to bastille

In order to make Bastille functional and avoid errors
regarding distros, if not any given distro is identified,
yocto-standard distro is added to the distro variable
in Bastille.

Fixed also some warnings regarding defined statements
in API.pm.

Upstream Status: Inappropriate [No upstream maintenance]

Signed-off-by: Andrei Dinu <andrei.adrianx.dinu@intel.com>

Signed-off-by: Anne Mulhern <mulhern@yoctoproject.org>

---
 Bastille/API.pm |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

Index: Bastille/Bastille/API.pm
===================================================================
--- Bastille.orig/Bastille/API.pm	2008-09-14 19:56:53.000000000 -0400
+++ Bastille/Bastille/API.pm	2013-08-21 08:55:26.715950001 -0400
@@ -445,8 +445,8 @@
 		$release=`/usr/bin/uname -sr`;
 	    }
 	    else {
-	 	print STDERR "$err Could not determine operating system version!\n";
-		$distro="unknown";
+                print STDERR "$err Could not determine operating system version!\n";
+		$distro="unknown"
             }
 
 	    # Figure out what kind of system we're on.
@@ -1284,7 +1284,7 @@
 
     my $sumFile = &getGlobal('BFILE',"sum.csv");
 
-    if ( defined %GLOBAL_SUM ) {
+    if ( %GLOBAL_SUM ) {
 
 	open( SUM, "> $sumFile") or &B_log("ERROR","Unable to open $sumFile for write.\n$!\n");
 
@@ -1318,7 +1318,7 @@
     my $file = $_[0];
     my $cksum = &getGlobal('BIN',"cksum");
 
-    if (not(defined(%GLOBAL_SUM))) {
+    if (not(%GLOBAL_SUM)) {
         &B_read_sums;
     }
 
@@ -1375,7 +1375,7 @@
 sub B_isFileinSumDB($) {
     my $file = $_[0];
 
-    if (not(defined(%GLOBAL_SUM))) {
+    if (not(%GLOBAL_SUM)) {
         &B_log("DEBUG","Reading in DB from B_isFileinSumDB");
         &B_read_sums;
     }
